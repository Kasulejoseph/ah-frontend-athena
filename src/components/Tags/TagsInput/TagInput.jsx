import React, { Component } from "react";
import TagsInput from "react-tagsinput";
import PropTypes from "prop-types";

import "react-tagsinput/react-tagsinput.css";

class TagInput extends Component {
  state = { tags: [] };

  handleChange = tags => {
    const { handleTags } = this.props;
    this.setState({ tags });
    handleTags(tags, this.state);
  };

  render() {
    const { tagList } = this.props;
    return <TagsInput value={tagList} onChange={this.handleChange} />;
  }
}

TagInput.propTypes = {
  handleTags: PropTypes.func,
  tagList: PropTypes.shape({})
};

TagInput.defaultProps = {
  handleTags: () => {},
  tagList: []
};

export default TagInput;
