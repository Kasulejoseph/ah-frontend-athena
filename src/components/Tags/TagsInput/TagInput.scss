.react-tagsinput-tag {
  background-color: #1aa1e2 !important;
  border-radius: 14px !important;
  border: 0px !important;
  color: #fff !important;
  font-size: 15px !important;
  font-weight: bolder !important;
}
