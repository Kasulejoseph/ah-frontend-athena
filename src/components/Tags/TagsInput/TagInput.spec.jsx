import React from "react";
import { shallow } from "enzyme";
import TagInput from "./TagInput";

let tags = ["magic", "angular", "andela"];
const props = {
  tags
};

describe("<TagInput />", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TagInput {...props} {...tags} />);
  });

  it("matches snapshort", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange ", () => {
    const instance = wrapper.instance();
    expect(
      instance.handleChange({
        target: { name: "title", value: "anything" },
        preventDefault: jest.fn
      })
    );
  });
});
