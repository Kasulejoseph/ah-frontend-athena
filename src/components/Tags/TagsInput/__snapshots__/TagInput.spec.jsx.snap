// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagInput /> matches snapshort 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagInput
    0="magic"
    1="angular"
    2="andela"
    handleTags={[Function]}
    tagList={Array []}
    tags={
      Array [
        "magic",
        "angular",
        "andela",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "addKeys": Array [
        9,
        13,
      ],
      "addOnBlur": false,
      "addOnPaste": false,
      "className": "react-tagsinput",
      "disabled": false,
      "focusedClassName": "react-tagsinput--focused",
      "inputProps": Object {},
      "maxTags": -1,
      "onChange": [Function],
      "onlyUnique": false,
      "pasteSplit": [Function],
      "preventSubmit": true,
      "removeKeys": Array [
        8,
      ],
      "renderInput": [Function],
      "renderLayout": [Function],
      "renderTag": [Function],
      "tagDisplayProp": null,
      "tagProps": Object {
        "className": "react-tagsinput-tag",
        "classNameRemove": "react-tagsinput-remove",
      },
      "validationRegex": /\\.\\*/,
      "value": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addKeys": Array [
          9,
          13,
        ],
        "addOnBlur": false,
        "addOnPaste": false,
        "className": "react-tagsinput",
        "disabled": false,
        "focusedClassName": "react-tagsinput--focused",
        "inputProps": Object {},
        "maxTags": -1,
        "onChange": [Function],
        "onlyUnique": false,
        "pasteSplit": [Function],
        "preventSubmit": true,
        "removeKeys": Array [
          8,
        ],
        "renderInput": [Function],
        "renderLayout": [Function],
        "renderTag": [Function],
        "tagDisplayProp": null,
        "tagProps": Object {
          "className": "react-tagsinput-tag",
          "classNameRemove": "react-tagsinput-remove",
        },
        "validationRegex": /\\.\\*/,
        "value": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
